src/test/java/tests/TickingTimerTests.java

```diff
@@ -86,4 +88,61 @@ public void triggerTest() {
             }
         }
     }
+
+
+    //This method is to test the changePeriod method in TickingTimer class. The changePeriod method is called 0.1sec
+    //before timeout.
+    @Test
+    public void changePeriodTest() {
+        final ArrayList<Integer> ticks = new ArrayList<>();
+
+        final TickingTimer tickingTimer = new TickingTimer("test3", 5, (Integer i) -> {}, () -> {
+            tickingTimerStatus = true;
+            ticks.clear();
+        }, TimeUnit.SECONDS);
+
+        tickingTimer.start();
+
+        delay(3.9); //TickingTimer initial timeout is 1 sec less.
+
+        if (hasTimeout()) {
+            fail();
+        } else {
+            tickingTimer.changePeriod(3);
+        }
+
+        delay(1.5);
+
+        //At t=1.5s, timeout should not be called yet.
+        if (hasTimeout()) {
+            fail();
+        }
+
+        delay(1.6);
+        //At t=3.1s, timeout should be called already.
+        if (hasTimeout()) {
+            tickingTimerStatus = false; //Reset the flag.
+        } else {
+            fail();
+        }
+
+        //Second iteration after calling changePeriod method.
+        delay(2);
+
+        //At t=2.1s, timeout should not be called yet.
+        if (hasTimeout()) {
+            fail();
+        }
+
+        delay(1.5);
+
+        //At t=3.6s, timeout should be called already.
+        if (!hasTimeout()) {
+            fail();
+        }
```

Comment:
I have this part of the code that need to reset the status. So I thought I should be consistent for the rest. 

>   if (hasTimeout()) {
            tickingTimerStatus = false; //Reset the flag.
        } else {
            fail();
        }

>